### PR TITLE
Caseflow 5066 sync changehearingrequesttypetask with representative change

### PR DESCRIPTION
**Pull request by j-n-t-l**
Resolves #[CASEFLOW-5066](https://vajira.max.gov/browse/CASEFLOW-5066)
Description
As a newly appointed rep I need to the ability Veteran with a new representative - when the representative of an appeal changes, the change hearing request type task is reassigned to the new representative

Update Appellant Representation Job needs to be updated to ensure the ChangeHearingRequestTypeTask for a given Appeal is assigned to the correct VSO user when there is a change in representation for the appeal.

Acceptance Criteria

When the representative of an appeal changes:
- [ ] the change hearing request type task is unassigned from the prior representative (can no longer convert to virtual)
- [ ] the change hearing request type task is assigned to the new representative (can convert to virtual)

### CODE REVIEW

**Review by ksross-va — APPROVED** [review]
LGTM!

Just for traceability, this PR is a slight revision to this one, due to a change noticed after the merge: https://github.com/department-of-veterans-affairs/caseflow/pull/17212

**Review by ksross-va — CHANGES_REQUESTED** [review]
Spec tests are failing now that the `after_save` and `after_destroy` are running. 

spec/workflows/initial_tasks_factory_spec.rb will need to be updated

**Comment by ksross-va** [comment]
A note for anyone else reviewing this PR - ignore these file changes:

- app/controllers/idt/api/v2/appeals_controller.rb
- app/models/decision_document.rb
- lib/tasks/ci.rake
- spec/controllers/idt/api/v2/appeals_controller_spec.rb
- spec/jobs/dispatch_email_job_spec.rb
- spec/models/decision_document_spec.rb

These were pulled in from master at a point when master was ahead of the feature branch, so those are not actually changes made by this PR. These changes _are_ now in the feature branch so I didn't think it worthwhile to revert.

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 1]
```diff
@@ -70,6 +72,43 @@ def update_from_params(params, user)
     end
   end
 
+  # rubocop:disable Metrics/MethodLength
```
Comment (j-n-t-l) [inline app/models/tasks/change_hearing_request_type_task.rb:75]: start of my changes

Comment (j-n-t-l) [inline app/models/tasks/change_hearing_request_type_task.rb:75]: start of my changes

File: .codeclimate.yml [hunk 2]
```diff
@@ -57,6 +57,7 @@ plugins:
     - 'spec/controllers/idt/api/appeals_controller_spec.rb'
     - 'spec/controllers/tasks_controller_spec.rb'
     - 'spec/controllers/decision_reviews_controller_spec.rb'
+    - 'app/models/tasks/change_hearing_request_type_task.rb'
```
Comment (j-n-t-l) [inline .codeclimate.yml:60]: my change

Comment (j-n-t-l) [inline .codeclimate.yml:60]: my change

File: app/models/bgs_power_of_attorney.rb [hunk 3]
```diff
@@ -15,6 +15,8 @@ class BgsPowerOfAttorney < CaseflowRecord
   before_save :update_cached_attributes!
   after_save :update_ihp_task, if: :update_ihp_enabled?
   after_destroy :update_ihp_task, if: :update_ihp_enabled?
+  after_save :update_changehearingrequesttype_task, if: :update_changehearingrequesttype_enabled?
```
Comment (j-n-t-l) [inline app/models/bgs_power_of_attorney.rb:18]: everything in this file are my changes

Comment (j-n-t-l) [inline app/models/bgs_power_of_attorney.rb:18]: everything in this file are my changes

File: app/views/queue/index.html.erb [hunk 4]
```diff
@@ -37,6 +37,7 @@
       poa_button_refresh: FeatureToggle.enabled?(:poa_button_refresh, user: current_user),
```
Comment (j-n-t-l) [inline app/views/queue/index.html.erb:37]: this file has my changes

Comment (j-n-t-l) [inline app/views/queue/index.html.erb:37]: this file has my changes

File: spec/models/bgs_power_of_attorney_spec.rb [hunk 5]
```diff
@@ -357,6 +357,37 @@
       end
     end
```
Comment (j-n-t-l) [inline spec/models/bgs_power_of_attorney_spec.rb:359]: this file has my changes

Comment (j-n-t-l) [inline spec/models/bgs_power_of_attorney_spec.rb:359]: this file has my changes

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 6]
```diff
@@ -72,6 +72,43 @@ def update_from_params(params, user)
...
+  # rubocop:disable Metrics/AbcSize
+  def self.update_to_new_poa(appeal)
+    new_task_count = 0
+    closed_task_count = 0
+
+    tasks_to_sync = appeal.tasks.open.where(
+      type: [ChangeHearingRequestTypeTask.name],
+      assigned_to_type: Organization.name
```
Comment (j-n-t-l) [inline app/models/tasks/change_hearing_request_type_task.rb:83]: Should be assigned_to_type: User.name

File: .codeclimate.yml (hunk 2 above)
Comment (ajspotts) [inline .codeclimate.yml:60]:
```suggestion
```

**Pull request closed without merging**